mne/preprocessing/stim.py

```diff
@@ -67,3 +91,92 @@ def eliminate_stim_artifact(raw, events, event_id, tmin=-0.005,
+        The id of the events generating the stimulation artifacts.
+        No need when inst is epochs or evoked
+    tmin : float
+        Start time of the interpolation window in seconds.
+    tmax : float
+        End time of the interpolation window in seconds.
+    mode : 'linear' | 'window'
+        way to fill the artifacted time interval.
+        'linear' does linear interpolation
+        'window' applies a (1 - hanning) window.
+    copy : bool
+        If False inst is cropped in place.
+
+    Returns
+    -------
+    inst : instance of evoked or epochs
+        instance with modified data
+    """
+    if copy:
+        inst = inst.copy()
+    s_start = int(np.ceil(inst.info['sfreq'] * tmin))
+    s_end = int(np.ceil(inst.info['sfreq'] * tmax))
+    if (s_end - s_start - 4) < 0:
+        raise ValueError('Time between tmin and tmax is too short'
+                         ' to fix artifact. Input longer values')
+    window = None
+    if mode == 'window':
+        window = _get_window(s_start, s_end)
+    ch_names = inst.info['ch_names']
+    picks = pick_channels(ch_names, ch_names)
+
+    if isinstance(inst, Raw):
+        if inst.preload is False:
+            raise RuntimeError('Modifying data of Raw is only supported '
+                               'when preloading is used. Use preload=True '
+                               '(or string) in the constructor.')
+        events_sel = (events[:, 2] == event_id)
+        event_start = events[events_sel, 0]
+        if mode is 'linear':
+            data, _ = inst[picks, :]
+        elif mode is 'window':
+            data, _ = inst[:, :]
+        for k in range(len(event_start)):
+            first_samp = int(event_start[k]) - inst.first_samp + s_start
+            last_samp = int(event_start[k]) - inst.first_samp + s_end
+            _fix_artifact(inst, data, window, picks, first_samp, last_samp, mode)
+
+    elif isinstance(inst, Epochs):
+        if inst.preload is False:
+            raise RuntimeError('Modifying data of Epochs is only supported '
+                               'when preloading is used. Use preload=True '
+                               'in the constructor.')
+        if inst.reject:
+            raise RuntimeError('Reject is already applied. Use reject=False '
+                               'in the constructor.')
+        e_start = int(np.ceil(inst.info['sfreq'] * inst.tmin))
+        first_samp = s_start - e_start
+        last_samp = s_end - e_start
+        data = inst.get_data()[:, picks, :]
+        k = 0
+        for epoch in data:
+            _fix_artifact(inst._data[k], epoch, window, picks, first_samp, last_samp, mode)
+            k += 1
+
+    elif isinstance(inst, Evoked):
+        first_samp = s_start - inst.first
+        last_samp = s_end - inst.first
+        data = inst.data
+        _fix_artifact(inst.data, data, window, picks, first_samp, last_samp, mode)
+
+    else:
+        raise TypeError('Not a Epochs or Evoked or Raw')
```

Comment:
nitpick but I'd say `Not a Raw or Epochs or Evoked` as that is the typical order of processing